Sources/StreamChat/Repositories/SyncRepository.swift

```diff
@@ -64,6 +63,32 @@ class SyncRepository {
         self.database = database
         self.apiClient = apiClient
     }
+    
+    deinit {
+        cancelRecoveryFlow()
+    }
+    
+    func syncLocalState(completion: @escaping () -> Void) {
+        cancelRecoveryFlow()
+        
+        getUser { [weak self] in
+            guard let currentUser = $0 else {
+                log.error("Current user must exist", subsystems: .offlineSupport)
+                completion()
```

Comment:
This errors but nothing is done?

Reply:
Yep, it's weird but I decided not to touch the API of `SyncRepository`, just adjust the implementation. 

We can update the completion closure in `syncLocalState` to receive error but the thing is we don't interrupt the reconnection flow when some operation fails with an unrecoverable error we just move on to the next step.

Unless we propagate error from operation somehow, adding `Error` to completion won't change much because even if recovery flow has failed, completion would be called with `nil` error.